src/core/presentation_definition.rs

```diff
@@ -223,23 +223,16 @@ impl PresentationDefinition {
             .filter(|field| field.is_required())
             .all(|field| {
                 match field.filter.as_ref().map(|f| f.validator()) {
-                    Some(validator) => {
-                        let is_valid = field
-                            .path
-                            .iter()
-                            // NOTE: Errors are ignored to allow other paths to
-                            // be checked. Interested in whether there is at least
-                            // one valid path.
-                            //
-                            // An empty iterator will return false on an any() call.
-                            .flat_map(|path| path.query(credential))
-                            // NOTE: This is currently assuming that if any of the paths are a match
-                            // to the credential, then the validation is, at least partially, successful,
-                            // and the credential may satisfy the presentation definition.
-                            .any(|value| validator.validate(value).is_ok());
-
-                        is_valid
-                    }
+                    Some(validator) => field
+                        .path
+                        .iter()
+                        .flat_map(|path| path.query(credential))
+                        .any(|value| match value {
+                            serde_json::Value::Array(vals) => {
+                                vals.iter().any(|val| validator.validate(val).is_ok())
+                            }
+                            _ => validator.validate(value).is_ok(),
+                        }),
```

Comment:
Just realized this introduces a bug if the constraint _is_ for an array, so probably need to do this:

```suggestion
                        .any(|value| {
                            match value {
                                // If the value is an array, test the constraint against each array element.
                                serde_json::Value::Array(vals) => {
                                    if vals.iter().any(|val| validator.validate(val).is_ok()) {
                                        return true;
                                    }
                                },
                                _ => (),
                            }
                            validator.validate(value).is_ok()
                        }),
```